### PR TITLE
#33374: Add linter rules, fix. Add contextmenu

### PR DESCRIPTION
**Pull request by m0sk1t**
## USER STORY 33374

- Add context menu
- Upgrade packages & linter
- Fix linter issues

tornado_menu

### CODE REVIEW

**Comment by m0sk1t** [comment]
@uve please take a look

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "@types/karma": "3.0.0",
```
Comment (uve) [inline package.json:27]: Increase version number in:

pbiviz.json
package.json
package-lock.json

File: package.json [hunk 2]
```diff
@@ -48,7 +49,7 @@
     "less-loader": "^4.1.0",
     "lodash": "^4.17.15",
     "powerbi-visuals-api": "2.6.1",
```
Comment (uve) [inline package.json:51]: Update all dependencies

File: src/TornadoChart.ts [hunk 3]
```diff
@@ -29,52 +29,52 @@ import "./../style/tornadoChart.less";
 import "core-js/stable";
 import * as d3 from "d3";
 import * as _ from "lodash";
-import powerbi from "powerbi-visuals-api";
```
Comment (uve) [inline src/TornadoChart.ts:32]: keep the same import:
 **import powerbi from "powerbi-visuals-api";**

File: src/TornadoChart.ts [hunk 4]
```diff
@@ -29,52 +29,52 @@ import "./../style/tornadoChart.less";
...
+
+import IVisual = powerbiVisualsApi.extensibility.visual.IVisual;
+import IVisualHost = powerbiVisualsApi.extensibility.visual.IVisualHost;
+import ISelectionManager = powerbiVisualsApi.extensibility.ISelectionManager;
+import ISelectionId = powerbiVisualsApi.visuals.ISelectionId;
 
 import { dataViewObject, dataViewObjects } from "powerbi-visuals-utils-dataviewutils";
 
 import * as SVGUtil from "powerbi-visuals-utils-svgutils";
-import SVGManipulations = SVGUtil.manipulation;
```
Comment (uve) [inline src/TornadoChart.ts:66]: replace it by:
svgManipulation

File: src/TornadoChart.ts [hunk 5]
```diff
@@ -172,7 +172,15 @@ export class TornadoChart implements IVisual {
         categoriesFillColor: "#777"
     };
 
-    public static converter(
+    private static buildIdentitySelection(hostService, category, i, values, columnGroup, measureName): ISelectionId {
```
Comment (uve) [inline src/TornadoChart.ts:175]: Add types to each of arguments

File: src/TornadoChart.ts [hunk 6]
```diff
@@ -172,7 +172,15 @@ export class TornadoChart implements IVisual {
...
+        return hostService.createSelectionIdBuilder()
+            .withCategory(category, i)
+            .withSeries(values, columnGroup)
+            .withMeasure(measureName)
+            .createSelectionId();
+    }
+
+    public static CONVERTER(
```
Comment (uve) [inline src/TornadoChart.ts:183]: do not use capital cases for the function:
keep it as "converter"

File: src/tooltipBuilder.ts [hunk 7]
```diff
@@ -49,7 +49,6 @@ export function createTooltipInfo(
     localizationManager: ILocalizationManager,
     value?: any,
     seriesIndex?: number,
-    categoryIndex?: number,
```
Comment (uve) [inline src/tooltipBuilder.ts:52]: why did you remove categoryIndex here?

File: test/TornadoChartBuilder.ts [hunk 8]
```diff
@@ -101,7 +101,7 @@ export class TornadoChartBuilder extends VisualBuilderBase<VisualClass> {
         isGrouped: boolean,
         columnGroup: DataViewValueColumnGroup): TornadoChartSeries {
 
-        return VisualClass.parseSeries(
+        return VisualClass.PARSE_SERIES(
```
Comment (uve) [inline test/TornadoChartBuilder.ts:104]: do no rename the function name to capital cases

File: test/TornadoData.ts [hunk 9]
```diff
@@ -24,9 +24,9 @@
  *  THE SOFTWARE.
  */
 
-import powerbi from "powerbi-visuals-api";
```
Comment (uve) [inline test/TornadoData.ts:27]: Please keep:
import powerbi from "powerbi-visuals-api";

File: test/helpers/helpers.ts [hunk 10]
```diff
@@ -70,7 +70,7 @@ export function getHexColorFromNumber(value: number): string {
     const hexColor: string = value.toString(16).toUpperCase(),
         color: string = hexColor.length === 6
             ? hexColor
-            : `${_.range(0, 6 - hexColor.length, 0).join("")}${hexColor}`;
```
Comment (uve) [inline test/helpers/helpers.ts:73]: Does it work?
How?

File: test/visualTest.ts [hunk 11]
```diff
@@ -24,24 +24,24 @@
  *  THE SOFTWARE.
  */
 
-import powerbi from "powerbi-visuals-api";
```
Comment (uve) [inline test/visualTest.ts:27]: keep old name:
import powerbi from "powerbi-visuals-api";

File: src/TornadoChart.ts (hunk 6 above)
Comment (uve) [reply src/TornadoChart.ts:183]: do you really need to make this function public?

File: package.json [hunk 12]
```diff
@@ -48,7 +49,7 @@
     "less-loader": "^4.1.0",
     "lodash": "^4.17.15",
```
Comment (uve) [inline package.json:50]: Please also remove lodash package completely.
If any lodash methods are required, install them separately, or via lodash-es

File: src/TornadoChart.ts [hunk 13]
```diff
@@ -1281,24 +1263,20 @@ export class TornadoChart implements IVisual {
     }
 
     private enumerateCategories(settings: TornadoChartSettings): VisualObjectInstance[] {
-        let position: string = dataViewObject.getValue<string>(
```
Comment (uve) [inline src/TornadoChart.ts:1284]: keep position as it was, sometimes we use it for debugging.

**Comment by uve** [comment]
Tests are failed.

File: src/tooltipBuilder.ts (hunk 7 above)
Comment (m0sk1t) [reply src/tooltipBuilder.ts:52]: This argument isn't used in function body

**Comment by m0sk1t** [comment]
@uve please take a look again)
